code/go/0chain.net/smartcontract/zcnsc/mint.go

```diff
@@ -41,6 +41,11 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
 		return
 	}
 
+	// if number of slices exceeds limits the check only withing required range
+	if len(payload.Signatures) > gn.MaxSignatures {
```

Comment:
Threshold should also be considered, imagine `gn.MaxSignatures < threshold`. Thiss could happen when we have lot of authorizers registered, suppose 200, with the threshold of 0.7, the required signatures would be 140, which is greater than the default max signatures of 100. So perhaps the max signatures config is not necessary, we can use threshold as the max number of signatures we're going to check.

Reply:
we can use ``if len(payload.Signatures) > max(gn.MaxSignatures, threshold)`` but frankly I'd rejected all mint transactions with more than `gn.MaxSignatures` and didn't slice it even

Reply:
The `gn.MaxSignatures` does not scale, as we can register new authorizers without limitation, though we will control how many authorizers would be allowed, but then it is the `MaxAuthorizers * threshold` that we should check instead of the `gn.MaxSignatures`.